### PR TITLE
Cleanup E0749

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @pickfire

### CODE REVIEW

**Review by JohnTitor — APPROVED** [review]
Looks good!

**Comment by pickfire** [comment]
@GuillaumeGomez Looks good to me but this feels familiar to the one you changed earlier, is it duplicated?

File: src/librustc_error_codes/error_codes/E0749.md [hunk 1]
```diff
@@ -1,4 +1,19 @@
...
+```compile_fail,E0749
+# #![feature(negative_impls)]
+trait MyTrait {
+    type Foo;
+}
+
+impl !MyTrait for u32 {
+    type Foo = i32; // error!
+}
+# fn main() {}
```
Comment (pickfire) [inline src/librustc_error_codes/error_codes/E0749.md:14]: Do we suggested how to fix this? Is it clear that the user should remove the type?

Comment (JohnTitor) [reply src/librustc_error_codes/error_codes/E0749.md:14]: Hm it's verbose for me. "item" here is ambiguous a bit?

File: src/librustc_error_codes/error_codes/E0749.md [hunk 2]
```diff
@@ -1,4 +1,19 @@
-Negative impls are not allowed to have any items. Negative impls
-declare that a trait is **not** implemented (and never will be) and
-hence there is no need to specify the values for trait methods or
-other items.
+An item was added on a negative impl.
```
Comment (pickfire) [inline src/librustc_error_codes/error_codes/E0749.md:1]: @JohnTitor I think the "item" here like you said means "type". Should we change it?
```suggestion
An associated type was added on a negative impl.
```

Comment (JohnTitor) [reply src/librustc_error_codes/error_codes/E0749.md:1]: Well, "item" means "associated items" so I don't think we should change it.

File: src/librustc_error_codes/error_codes/E0749.md (hunk 1 above)
Comment (GuillaumeGomez) [reply src/librustc_error_codes/error_codes/E0749.md:14]: The explanation below isn't enough?

**Comment by Dylan-DPC** [comment]
merging this

@bors r+ rollup

File: src/librustc_error_codes/error_codes/E0749.md (hunk 1 above)
Comment (pickfire) [reply src/librustc_error_codes/error_codes/E0749.md:14]: Just enough but it is not as straightforward.

Comment (GuillaumeGomez) [reply src/librustc_error_codes/error_codes/E0749.md:14]: Hum... I'm not sure to see what's missing so could please open a PR with your vision? (let's switch position for once! :p)

Set me as reviewer on it so we can move forward quickly! :)

**Pull request merged**